### PR TITLE
Make `waitUntilSlot` to wait for any high slot

### LINKED ISSUE #1466: `waitNSlots` can not wait for a slot outside of current eras boundaries
The problem arise on large waiting intervals.
However, it seems resolvable.

## Implementation

Pseudocode:
```hs
waitNSlots' 0 = pure unit
waitNSlots' n = do
  highestSlot <- highestSlotInEraSummaries
  let (toWaitNow, toWaitAfter)
        | n <= highestSlot = (n,0)
        | otherwise = (highestSlot, n - highestSlot)
  waitNSlots toWaitNow
  waitNSlots' toWaitAfter
```

### PR DESCRIPTION
**Pull request by Renegatto**
Closes #1466.

### Pre-review checklist

- [x] All code has been formatted using our config (`make format`)
- [x] Any new API features or modification of existing behavior is covered as defined in our [test plan](https://github.com/Plutonomicon/cardano-transaction-lib/blob/develop/doc/test-plan.md)
- [x] The changelog has been updated under the `## Unreleased` header, using the appropriate sub-headings (`### Added`, `### Removed`, `### Fixed`), and the links to the appropriate issues/PRs have been included

### CODE REVIEW

File: src/Internal/Types/Interval.purs [hunk 1]
```diff
@@ -1,32 +1,20 @@
...
-      , StartTimeGreaterThanTime
-      , EndSlotLessThanSlotOrModNonZero
-      , CannotGetBigIntFromNumber'
-      , CannotGetBigNumFromBigInt'
-      )
-  , RelSlot(RelSlot)
-  , RelTime(RelTime)
+  , PosixTimeToSlotError(..)
+  , RelSlot(..)
+  , RelTime(..)
```
Comment (klntsky) [inline src/Internal/Types/Interval.purs:13]: Please undo these changes. We use explicit imports and exports.

File: test/Plutip/Contract.purs [hunk 2]
```diff
@@ -191,6 +192,14 @@ suite = do
         withKeyWallet alice ManyAssets.contract
 
   group "Contract interface" do
+    -- NOTE: plutip timeout must be increased for this test to pass.
```
Comment (klntsky) [inline test/Plutip/Contract.purs:195]: Please increase it

**Review by klntsky — CHANGES_REQUESTED** [review]

File: src/Internal/Types/Interval.purs (hunk 1 above)
Comment (Renegatto) [reply src/Internal/Types/Interval.purs]: Done.

File: test/Plutip/Contract.purs (hunk 2 above)
Comment (Renegatto) [reply test/Plutip/Contract.purs]: Done.

File: src/Internal/Contract/WaitUntilSlot.purs [hunk 3]
```diff
@@ -50,20 +51,39 @@ waitUntilSlot futureSlot = do
...
+            getEraSummaries =
+              liftAff queryHandle.getEraSummaries
+                >>= liftContractE
+          eraSummaries <- getEraSummaries
+          highestSlot <- liftContractM "Can't find any Era summary" $
+            highestEndSlotInEraSummaries eraSummaries
+          let
+            toWait = case highestSlot of
+              Just highestEndSlot
+                | Just waitThen <- futureSlot `sub` highestEndSlot
```
Comment (klntsky) [inline src/Internal/Contract/WaitUntilSlot.purs:64]: This subtraction is not needed, because futureSlot is the target slot we wait for. It can't be changed.

**Comment by klntsky** [comment]
I will re-create this PR from a branch in the repo for CI to run.
